packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -86,6 +88,22 @@ class HotRunner extends ResidentRunner {
       // Already computed.
       return true;
     }
+
+    try {
+      // Will return immediately if pubspec.yaml is up-to-date.
+      await pubGet(
```

Comment:
Can you confirm that this does print some appropriate message (so the user knows that a pub get is running)?

Reply:
Well.. it wasn't.  This turned out to be because hot reload calls `startProgress` on the logger, and then `pubGet` does the same, and the daemon code uses a custom logger that completely ignores nested progress messages.

I've changed it to print nested progress messages again.  You appear to have disabled them (https://github.com/flutter/flutter/pull/6677) to make the messages for hot reload and hot restart the same, however, so that might not be the right answer.  Other alternatives would be to add an argument to tell startProgress to not ignore specific messages, or to change pubGet to not start its own progress (but that's nice to have on the command line).

Reply:
The progress bar is an indeterminate bar on IntelliJ, and it does actually seem to work (it turns on the indeterminate progress bar in the status area when the reload starts, and turns it off when it's done). For this case, there are only a couple of steps (running pub get, and syncing to the device), but I imagine there could be some where it was a lot of steps flying by.

Of course, it looks fine in the IntelliJ "Run" console output, and on the regular command line.

Reply:
Sounds good! Two steps, where they both don't just flash by, isn't a concern for me. I would just get concerned if we had many very short steps - the text would flash by too quickly for users to read.